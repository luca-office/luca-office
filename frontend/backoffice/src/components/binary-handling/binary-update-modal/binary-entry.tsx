import {css} from "@emotion/react"
import * as React from "react"
import {Heading} from "shared/components"
import {HeadingLevel} from "shared/enums"
import {Children, CustomStyle, Flex, FontWeight, spacingTiny} from "shared/styles"
import {LucaI18nLangKey, useLucaTranslation} from "shared/translations"

export interface BinaryEntryProps extends CustomStyle, Children {
  readonly className?: string
  readonly headerKey: LucaI18nLangKey
}

export const BinaryEntry: React.FC<BinaryEntryProps> = ({children, className, customStyles, headerKey}) => {
  const {t} = useLucaTranslation()
  return (
    <div css={[styles.entry, customStyles]} className={className}>
      <Heading
        customStyles={styles.heading}
        fontWeight={FontWeight.Bold}
        level={HeadingLevel.h3}
        className={"binary-entry-header"}>
        {t(headerKey)}
      </Heading>
      <div className={"binary-entry-content"}>{children}</div>
    </div>
  )
}

const styles = {
  entry: css(Flex.column),
  heading: css({
    marginBottom: spacingTiny
  })
}
