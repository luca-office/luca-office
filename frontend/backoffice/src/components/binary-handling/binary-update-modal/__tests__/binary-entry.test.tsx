import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {BinaryEntry, BinaryEntryProps} from "../binary-entry"

const defaultProps: BinaryEntryProps = {
  className: "binary-entry",
  headerKey: "file__preview__label"
}

const getComponent = (props?: Partial<BinaryEntryProps>) => <BinaryEntry {...{...defaultProps, ...props}} />

describe("binary-entry", () => {
  it("renders correctly", () => {
    const component = getComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure", () => {
    const component = getComponent()
    const tree = shallow(component)
    expect(tree.find(".binary-entry-header")).toHaveLength(1)
    expect(tree.find(".binary-entry-content")).toHaveLength(1)
  })
})
