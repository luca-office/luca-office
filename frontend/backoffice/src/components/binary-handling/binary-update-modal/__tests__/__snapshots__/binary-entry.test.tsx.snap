// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`binary-entry renders correctly 1`] = `
<div
  className="binary-entry css-yhhejc-BinaryEntry"
>
  <div
    className="binary-entry-header css-13s2sfk-Heading"
  >
    file__preview__label
  </div>
  <div
    className="binary-entry-content"
  />
</div>
`;
